providers/account.rb

```diff
@@ -169,7 +169,24 @@ def home_ssh_dir_resource(exec_action)
 
 def authorized_keys_resource(exec_action)
   # avoid variable scoping issues in resource block
-  ssh_keys = Array(new_resource.ssh_keys)
+  ssh_keys = []
+  Array(new_resource.ssh_keys).each do |item|
+    # if item is a valid ssh key, copy it.
+    if item =~ /^(ssh-(dss|rsa|ed25519)|ecdsa-sha2-\w+) AAAA/
```

Comment:
Are you talking about the comments after the actual public key? (Usually: `User Name <user.name@example.com>`) or actual comment lines like `# comment`?
I'm not sure what you're trying to catch with `(\s+)?#`.

My regex will match the first comment-type. The manpage allows comment-lines starting with `#`, so I guess it's valid to support them as well, even though I think in an automated environment comments should belong into the infrastructure-as-code respository.

This would then probably do the trick:
```ruby
/^(#|(ssh-(dss|rsa|ed25519)|ecdsa-sha2-\w+) AAAA)/
```